frontend/apps/app/middleware.ts

```diff
@@ -9,12 +9,15 @@ export async function updateSession(request: NextRequest) {
     return NextResponse.next()
   }
 
-  const migrationEnabled = await migrationFlag()
-
-  if (!migrationEnabled) {
-    const url = request.nextUrl.clone()
-    url.pathname = '/'
-    return NextResponse.redirect(url)
+  // Skip migration flag check in production environment
+  if (process.env.VERCEL_ENV !== 'production') {
+    const migrationEnabled = await migrationFlag()
+
+    if (!migrationEnabled) {
+      const url = request.nextUrl.clone()
+      url.pathname = '/'
+      return NextResponse.redirect(url)
+    }
```

Comment:
You want to keep the flag in the preview environment, correct? Perhaps the intention is to forbid logging in as a test user?
If we are going to disable the flag in the production environment, it might be more valuable to remove the flag altogether and make the Preview environment accessible only to the development team with Vercel Authentication. What do you think?

ref: [Vercel Authentication](https://vercel.com/docs/deployment-protection/methods-to-protect-deployments/vercel-authentication)

Reply:
@hoshinotsuyoshi yeah!! go ahead!